app/jobs/cron_job.rb

```diff
@@ -70,4 +70,16 @@ def perform
       Admins::SystemMailer.rdv_events_stats.deliver_later
     end
   end
+
+  class UpdateAgentToBeInformedRdvCountJob < CronJob
+    # At 04:11 AM, Monday through Friday
+    self.cron_expression = "11 4 * * 1-5"
```

Comment:
Ça veut dire que la bannière va rester dans le nez des agents toute la journée, même s’ils ont tout fait en arrivant. C’est un peu frustrant, non?

Il y a un [mécanisme similaire](https://guides.rubyonrails.org/association_basics.html#options-for-belongs-to-counter-cache) dans Rails, mais qui n’existe que pour les relations one-to-many (avec un belongs_to). En gros, ça rajoute un hook du côté “many” pour aller mettre à jour la colonne du côté “one”. Ici on pourrait faire la même chose: un hook dans `Rdv`, qui à chaque changement va modifier la valeur d’une colonne `rdv_count` des `Agent`s du Rdv. L’autre avantage, c’est que ça modifie le `updated_at` du `Agent`, et qu’on peut donc cacher le partial `rdv_a_renseigner` avec l’agent comme clé de cache.

Reply:
Ah oui, c'est pas mal aussi... 

Après, on a mis « toutes les nuits » on pourrait aussi le faire tourner plus souvent.

Je ne sais pas choisir entre hook et cron. Les deux me vont je crois. Je n'ai pas d'élément qui me permette de décider.

Par contre, je me dit que ça serait chouette d'être cohérent (et définir une orientation)